messaging-client-databus/src/main/java/com/inmobi/messaging/consumer/databus/DatabusConsumer.java

```diff
@@ -88,6 +85,10 @@
   public static String clusterNamePrefix = "databusCluster";
   private Boolean readFromLocalStream;
   private int numList = 0;
+  private  List<String> collectorsWithReaders = new ArrayList<String>();
```

Comment:
Discussed with @rajubairishetti . Here is the summarized approach:

1. Have a separate 'HashSet<PartitionId> newPartitions' in DatabusConsumer class.
2. Inside createPartitionReaders(), each collector-cluster entry will be first looked up in readers map. If not found, we first add all new entries in newPartitions set.
3. Have a separate startNewPartitionReaders() where we iterate over newPartitions hashset; first try to start each partition reader, if it succeeds then we add it to readers map. Else we throw IOException. In the finally block, we clear the hashset.
4. Timer task will simply call createPartitionReaders() and startNewPartitionReaders().